### PR TITLE
Describe optimization parameters in docs

### PR DESCRIPTION
**Pull request by ibrahim-shehzad**
Add explanation for the fields in `OptimizationParameters`, in particular `max_gamma` and `max_backjumps`.

### CODE REVIEW

**Comment by ibrahim-shehzad** [comment]
@garrison I went ahead and added some details but they are all in the docstring. I wonder if that is where we want to put it.

**Comment by garrison** [comment]
> I wonder if that is where we want to put it.

It'd be nice to get the descriptions in the "Attributes" table on [the page of the sphinx build](https://qiskit-extensions.github.io/circuit-knitting-toolbox/stubs/circuit_knitting.cutting.OptimizationParameters.html#circuit_knitting.cutting.OptimizationParameters).

If you format the descriptions as comments that start with `#:` like on the following lines, does that work?

https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/7029c106d7e807e6b27aaca70f54152463e5d47c/circuit_knitting/cutting/qpd/weights.py#L38-L45

**Comment by ibrahim-shehzad** [comment]
>If you format the descriptions as comments that start with #: like on the following lines, does that work?

It does ([f4a64b0](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/615/commits/f4a64b09acff3f20777e1219b830f7d14293dac9)). I did not get it to work with multiple sentences and/or multiple lines wasn't straightforward though so I kept everything in one sentence.

File: circuit_knitting/cutting/automated_cut_finding.py [hunk 1]
```diff
@@ -139,12 +139,25 @@ def find_cuts(
...
+    nevertheless returns the result of a greedy best first search, which gives an *upper-bound* on gamma.
+    """
+
+    #: The seed to use when initializing Numpy random number generators in the best first search priority queue.
     seed: int | None = OptimizationSettings().seed
+
+    #: Maximum allowed value of gamma which, if exceeded, forces the search to terminate.
     max_gamma: float = OptimizationSettings().max_gamma
+
+    #: Maximum number of backjumps that can be performed before the search is forced to terminate; setting it to None implies that no such restrictions is placed.
```
Comment (garrison) [inline circuit_knitting/cutting/automated_cut_finding.py:154]:
```suggestion
    #: Maximum number of backjumps that can be performed before the search is forced to terminate; setting it to ``None`` implies that no such restriction is placed.
```

File: circuit_knitting/cutting/automated_cut_finding.py [hunk 2]
```diff
@@ -139,12 +139,25 @@ def find_cuts(
...
 @dataclass
 class OptimizationParameters:
-    """Specify parameters that control the optimization."""
+    """Specify parameters that control the optimization.
 
+    If either of the constraints specified by ``max_backjumps`` or ``max_gamma`` are exceeded, the search terminates but
+    nevertheless returns the result of a greedy best first search, which gives an *upper-bound* on gamma.
+    """
+
+    #: The seed to use when initializing Numpy random number generators in the best first search priority queue.
```
Comment (garrison) [inline circuit_knitting/cutting/automated_cut_finding.py:148]: Optional suggestion

```suggestion
    #: The seed to use when initializing Numpy random number generators in the best-first-search priority queue.
```

**Review by garrison — APPROVED** [review]
Thank you - looks good to me! :rocket:

**Pull request merged**